pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -185,6 +200,90 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
+		if err != nil {
+			msg := fmt.Sprintf("Failed to remove VM %q with IP %q to IPList. Error: %+v", vm.Name, vm.Status.VmIp, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Successfully added VM IP %q to IPList for CnsFileAccessConfig request with name: %q on namespace: %q", vm.Status.VmIp, instance.Name, instance.Namespace)
+
+		volumeID, err := util.GetVolumeID(ctx, r.client, instance.Spec.PvcName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get volumeID from pvcName: %q. Error: %+v", instance.Spec.PvcName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		cnsVolumeID := cnstypes.CnsVolumeId{
+			Id: volumeID,
+		}
+		vSanFileShareNetPermissions := make([]vsanfstypes.VsanFileShareNetPermission, 0)
+
+		cfg, err := config.GetConfig()
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get config. Err: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// create a new dynamic client for config
+		dynamicClient, err := dynamic.NewForConfig(cfg)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to create client using config. Err: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// TODO:
+		// Identify the network provider (NSX-T or VDS). Assuming NSX-T configuration for now
+		tkgVMIP, err := cnsoperatorutil.GetTKGVMIP(ctx, r.vmOperatorClient, dynamicClient, vm.Namespace, vm.Name, true)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get external facing IP address for VM %q/%q. Err: %+v", vm.Namespace, vm.Name, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found tkg VMIP %q for VM %q in namespace %q", tkgVMIP, vm.Name, vm.Namespace)
+
+		// TODO:
+		// Fetch net permissions from config
+		vsanFileShareAccessType := vsanfstypes.VsanFileShareAccessTypeREAD_WRITE
+		vSanFileShareNetPermissions = append(vSanFileShareNetPermissions, vsanfstypes.VsanFileShareNetPermission{
+			Ips:         tkgVMIP,
+			Permissions: vsanFileShareAccessType,
+		})
+
+		cnsNFSAccessControlSpecList := make([]cnstypes.CnsNFSAccessControlSpec, 0)
+		cnsNFSAccessControlSpecList = append(cnsNFSAccessControlSpecList, cnstypes.CnsNFSAccessControlSpec{
+			Permission: vSanFileShareNetPermissions,
+			Delete:     true,
+		})
+
+		// Prepare the CnsVolumeACLConfigureSpec
+		cnsVolumeACLConfigSpec := cnstypes.CnsVolumeACLConfigureSpec{
+			VolumeId:              cnsVolumeID,
+			AccessControlSpecList: cnsNFSAccessControlSpecList,
+		}
+		r.volumeManager.ConfigureVolumeACLs(ctx, cnsVolumeACLConfigSpec)
```

Comment:
This only needs to be called if `ExternalIPtoClientVms[IP]` exists and `CnsFileAccessConfig.Spec.VirtualMachine.Name` is the only value in the `ExternalIPtoClientVms[IP]`